silkworm/silkrpc/core/blocks.cpp

```diff
@@ -103,7 +103,7 @@ boost::asio::awaitable<uint64_t> get_highest_block_number(const rawdb::DatabaseR
 }
 
 boost::asio::awaitable<uint64_t> get_latest_executed_block_number(const rawdb::DatabaseReader& reader) {
-    co_return co_await stages::get_sync_stage_progress(reader, stages::kExecution);
+    co_return co_await stages::get_sync_stage_progress(reader, stages::kFinish); // kFinish instead of kExecution
```

Comment:
Have we double checked if this modification is safe? 

Remember the reset_all case for estimate gas(). This is exactly the same pattern.

Is it safer to have another function and let the APIs we want to fix to call the new function?

In addition, I guess such change will break the unit test.